<?xml version="1.0"?>
<launch>
	<!-- Node: Local obstacles builder -->
	<node pkg="mrpt_local_obstacles" type="mrpt_local_obstacles_node" name="mrpt_local_obstacles_node" output="screen">
		<param name="source_topics_2dscan" value="laser1,laser2"/>
		<param name="show_gui" value="false"/>
	</node>
	<!-- Launch a MvSim simulator instance with one robot -->
	<arg name="world_file" default="$(find mvsim)/mvsim_tutorial/demo_1robot.world.xml" />
	<node pkg="mvsim" type="mvsim_node" name="mvsim_simulator" output="screen">
		<param name="world_file" value="$(arg world_file)"/>
		<param name="do_fake_localization" value="true"/>
	</node>
	<!-- Rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find mrpt_reactivenav2d)/tutorial/reactive_nav_demo_with_mvsim.rviz"/>


	<!-- Node: Pure Reactive Navigator -->
	<node pkg="mrpt_reactivenav2d" type="mrpt_reactivenav2d_node" name="mrpt_reactivenav2d_node" output="screen">
		<!-- *Important*: This external config file holds the most important navigation settings -->
		<param name="cfg_file_reactive" value="$(find mrpt_reactivenav2d)/tutorial/reactive2d_config.ini"/> 

		<!-- This is to allow the reactive responsive to RVIZ GUI commands -->
		<remap from="reactive_nav_goal" to="/move_base_simple/goal" />

		<!-- Enable logs to ~/.ros/reactivenav.logs , use navlog-viewer to open them -->
<!--		<param name="save_nav_log" value="true"/> -->

		<!-- Robot shape: Load from geometry_msgs::Polygon topic -->
		<param name="topic_robot_shape" value="/chassis_polygon" />
	</node>

</launch>

